### PR TITLE
Table column options added

### PR DESCRIPTION
**Pull request by xavijam**
In short:

screen shot 2016-06-22 at 15 25 06

- Add new column.
- Edit column type (+ confirmation modal).
- Edit column name (+ confirmation modal).
- Remove column (+ confirmation modal).
- Order by column (`ASC` and `DESC`).
- Generic modal view for the table itself (it will be used for table body operations in the next PR).

**CR** or at least a second pair of 👀  please: @nobuti  😄 
cc @piensaenpixel

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js [hunk 1]
```diff
@@ -10,16 +23,329 @@ module.exports = CoreView.extend({
...
+    this._menuView = new ContextMenuView({
+      className: menuOptionsClassName + ContextMenuView.prototype.className,
+      collection: collection,
+      itemTemplate: tableHeadOptionsItemTemplate,
+      itemView: TableHeadOptionsItemView,
+      triggerElementID: modelCID,
+      position: position
+    });
+
+    collection.bind('change:selected', function (menuItem) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js]: should not have an `off` anywhere?

File: lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js [hunk 2]
```diff
@@ -10,16 +23,329 @@ module.exports = CoreView.extend({
...
+      changeColumnTypeOperation({
+        columnModel: this.model,
+        newType: newType
+      });
+    } else {
+      this._modals.create(
+        function (modalModel) {
+          return new TableConfirmationView({
+            modalModel: modalModel,
+            template: require('./modals-templates/change-table-column-type.tpl'),
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js]: ❤️

File: lib/assets/javascripts/cartodb3/components/table/head/table-head-options-item-view.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          isOrderBy: this.model.get('isOrderBy'),
+          sortBy: this.model.get('sortBy'),
+          name: this.model.getName(),
+          val: this.model.getValue()
+        }
+      )
+    );
+
+    this.$el
+      .attr('data-val', this.model.getValue())
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/table/head/table-head-options-item-view.js]: why not `data('val', this.model.getValue())`?

File: lib/assets/javascripts/cartodb3/components/table/table-confirmation-modal-view.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+module.exports = CoreView.extend({
+  className: 'Dialog-content',
+
+  events: {
+    'click .js-confirm': '_onConfirm',
+    'click .js-cancel': '_onCancel'
+  },
+
+  initialize: function (opts) {
+    _.each(REQUIRED_OPTS, function (item) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/table/table-confirmation-modal-view.js]: ❤️

File: lib/assets/javascripts/cartodb3/components/table/table-confirmation-modal-view.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  _initBinds: function () {
+    $(document).bind('keydown', this._onKeyDown);
+  },
+
+  _disableBinds: function () {
+    $(document).unbind('keydown', this._onKeyDown);
+  },
+
+  _onKeyDown: function (ev) {
+    var keyCode = ev.keyCode || ev.which;
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/components/table/table-confirmation-modal-view.js]: No need of keyCode, I think jquery normalizes `which`

**Comment by nobuti** [comment]
💯

File: lib/assets/javascripts/cartodb3/components/table/head/table-head-options-item-view.js (hunk 3 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/table/head/table-head-options-item-view.js]: you are right!

File: lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/table/head/table-head-item-view.js]: Yep, applying just in case.

**Comment by xavijam** [comment]
Thanks man! waiting for travis frontend tests and merging :)

**Comment by xavijam** [comment]
Travis tests passed, merging!

**Pull request merged**